backend/src/bundles/auth/auth.ts

```diff
@@ -1,10 +1,10 @@
-import { userService } from '~/bundles/users/users.js';
+import { userRepository, userService } from '~/bundles/users/users.js';
 import { logger } from '~/common/logger/logger.js';
 
 import { AuthController } from './auth.controller.js';
 import { AuthService } from './auth.service.js';
 
-const authService = new AuthService(userService);
+const authService = new AuthService(userService, userRepository);
```

Comment:
I am not sure if we need to give authService direct access to userRepository since useRepository is accessible via userService? I am not sure though.

Reply:
> I am not sure if we need to give authService direct access to userRepository since useRepository is accessible via userService? I am not sure though.

I thinks you are right. You can create a method in userService which accesses userRepository since you alredy using userService in this class

Reply:
The main reason I imported repository is because I wanted to make findByEmail method inside user-service return just user object if we will need to use it, while inside auth-service we need userEntity with password hash

Reply:
Maybe in this case we can delete user-service and use only user-repository?

Reply:
> The main reason I imported repository is because I wanted to make findByEmail method inside user-service return just user object if we will need to use it, while inside auth-service we need userEntity with password hash

We can still achieve this with just userService exposed to authService.

You can return a UserEntity via useRepo and then return user Object by running UserEntity.toObject and returning result from service.

See [userService.findByEmail](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/88/files) sample flow.